gravitee-apim-repository/gravitee-apim-repository-jdbc/src/main/java/io/gravitee/repository/jdbc/management/JdbcEventRepository.java

```diff
@@ -408,38 +410,46 @@ public List<Event> findByOrganizationId(String organizationId) {
     }
 
     @Override
-    public Stream<EventToClean> findGatewayEvents(String environmentId) {
-        List<String> nonApiEventTypes = List.of(
-            "GATEWAY_STARTED",
-            "DEBUG_API",
-            "GATEWAY_STOPPED",
-            "PUBLISH_DICTIONARY",
-            "UNPUBLISH_DICTIONARY",
-            "START_DICTIONARY",
-            "STOP_DICTIONARY",
-            "PUBLISH_ORGANIZATION"
-        );
-        String sql =
-            """
-                SELECT e.id as id, ep.property_value as api
-                FROM %s e left join %s ev on e.id = ev.event_id left join %s ep on ep.event_id = e.id
-                WHERE type NOT IN (?) AND ev.environment_id = ? AND ep.property_key = 'api_id'
-                ORDER BY e.updated_at DESC
-            """.formatted(
-                    tableName,
-                    EVENT_ENVIRONMENTS,
-                    EVENT_PROPERTIES
-                );
-        String inClause = String.join(",", Collections.nCopies(nonApiEventTypes.size(), "?"));
-        sql = sql.replace("(?)", "(" + inClause + ")");
-        var params = new ArrayList<Object>(nonApiEventTypes);
-        params.add(environmentId);
-
-        return jdbcTemplate.queryForStream(
-            sql,
-            (rs, rowNum) -> new EventToClean(rs.getString("id"), rs.getString("api")),
-            params.toArray()
-        );
+    public Stream<EventToClean> findEventsToClean(String environmentId) {
+        log.debug("JdbcEventRepository.findEventsToClean({})", environmentId);
+        final List<Object> args = new ArrayList<>();
+        final StringBuilder builder = createSearchQueryBuilder();
+        addStringsWhereClause(Set.of(environmentId), "ev.environment_id", args, builder, false);
+
+        builder.append(" ORDER BY e.created_at DESC");
+
+        String sql = builder.toString();
+
+        return jdbcTemplate
+            .queryForStream(
+                sql,
+                (rs, rowNum) -> {
```

Comment:
Why don't you use `queryEvents()`? Because in your resultSet, you will get one row per property. So you will end with duplicated "event to clean"

Reply:
Seen together. It's ok